### PR TITLE
scotch: add missing dependency on zlib

### PR DESCRIPTION
**Pull request by johannesring**
Fixes issue #810.

### CODE REVIEW

File: pkgs/scotch/Make.inc.linux [hunk 1]
```diff
@@ -9,9 +9,9 @@ CAT		= cat
 CCS		= $(MPICC)
 CCP		= $(MPICC)
 CCD		= $(MPICC)
-CFLAGS		= -fPIC -O3 -DCOMMON_FILE_COMPRESS_GZ -DCOMMON_PTHREAD -DCOMMON_RANDOM_FIXED_SEED -DSCOTCH_RENAME -Drestrict=__restrict
+CFLAGS		= -I$(ZLIB_DIR)/include -fPIC -O3 -DCOMMON_FILE_COMPRESS_GZ -DCOMMON_PTHREAD -DCOMMON_RANDOM_FIXED_SEED -DSCOTCH_RENAME -Drestrict=__restrict
 CLIBFLAGS	=
-LDFLAGS		= -lz -lm -lrt
+LDFLAGS		= -L$(ZLIB_DIR)/lib -lz -lm -lrt
```
Comment (certik) [inline pkgs/scotch/Make.inc.linux]: I think you also need to set the rpath here.

**Comment by certik** [comment]
Otherwise +1.

File: pkgs/scotch/Make.inc.linux (hunk 1 above)
Comment (johannesring) [inline pkgs/scotch/Make.inc.linux]: Thanks @certik, but does it matter when we build a static library?

Comment (certik) [inline pkgs/scotch/Make.inc.linux]: Ah, it doesn't matter at all for a static library, I didn't realize you were building a static library.

+1 to merge then.

**Comment by johannesring** [comment]
Thanks!

**Pull request merged**